press/legacy_publishing/collection.py

```diff
@@ -55,17 +58,36 @@ def publish_legacy_book(model, metadata, submission, db_conn, changed=None):
 
     existing_shas = {filename: sha for filename, sha in shas}
 
-    # if the collection changed at all
-    existing_sha1 = existing_shas.get('collection.xml')
-    if existing_sha1 != produce_hashes_from_filepath(model.file)['sha1']:
-        raise CollectionChanged(model)
-    # OR if any of its resources changed
-    for res in model.resources:
-        if res.sha1 != existing_shas.get(res.filename):
-            raise CollectionChanged(model)
+    if changed is False and existing_shas['collection.xml'] \
+            == produce_hashes_from_filepath(model.file)['sha1']:
+        for res in model.resources:
+            if res.sha1 != existing_shas[res.filename]:
+                break  # publish!
+        else:  # collxml and all resources are identical to already published
+            raise Unchanged(model)
```

Comment:
Looks like this `CollectionChanged` and `Unchanged` code is already in, so my comments are just comments...

This use of state as exceptions smells funny. Just pointing that out. 😸